wen-restart/src/wen_restart.rs

```diff
@@ -264,12 +276,16 @@ pub(crate) fn find_heaviest_fork(
         .map(|(slot, _)| *slot)
         .collect::<Vec<Slot>>();
     slots.sort();
+
+    // The heaviest slot we selected will always be the last of the slots list, or root if the list is empty.
+    let heaviest_fork_slot = slots.last().map_or(root_slot, |x| *x);
+
     let mut expected_parent = root_slot;
-    for slot in slots {
+    for slot in &slots {
         if exit.load(Ordering::Relaxed) {
             return Err(WenRestartError::Exiting.into());
         }
-        if let Ok(Some(block_meta)) = blockstore.meta(slot) {
+        if let Ok(Some(block_meta)) = blockstore.meta(*slot) {
```

Comment:
Not sure if it is an unrecoverable error, but since we're already pulling the `SlotMeta` out, we could also check if the slot is full. Unless we'll be actively repairing / inserting shreds at this point in time, but this could be one of the causes of a bank not being frozen after replay below